File: api/src/main/java/com/ruby/rubia_server/core/entity/Company.java
package com.ruby.rubia_server.core.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.ruby.rubia_server.core.enums.CompanyPlanType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "companies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"departments"})
@EqualsAndHashCode(exclude = {"departments"})
public class Company {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false, unique = true)
    private String slug;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "contact_email")
    private String contactEmail;
    
    @Column(name = "contact_phone")
    private String contactPhone;
    
    @Column(name = "logo_url")
    private String logoUrl;
    
    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;
    
    @Enumerated(EnumType.ORDINAL)
    @Column(name = "plan_type")
    @Builder.Default
    private CompanyPlanType planType = CompanyPlanType.BASIC;
    
    @Column(name = "max_users")
    @Builder.Default
    private Integer maxUsers = 10;
    
    @Column(name = "max_whatsapp_numbers")
    @Builder.Default
    private Integer maxWhatsappNumbers = 1;
    
    @Column(name = "max_ai_agents")
    @Builder.Default
    private Integer maxAiAgents = 1;
    
    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonManagedReference("company-departments")
    private List<Department> departments;

    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonManagedReference("company-whatsapp")
    private List<WhatsAppInstance> whatsappInstances;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_group_id", nullable = false)
    @JsonBackReference("companygroup-companies")
    private CompanyGroup companyGroup;
}

File: api-dotnet/Rubia.Server/Entities/Company.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("companies")]
public class Company : BaseEntity
{
    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [Column("slug")]
    public string Slug { get; set; } = string.Empty;

    [Column("description", TypeName = "TEXT")]
    public string? Description { get; set; }

    [Column("contact_email")]
    public string? ContactEmail { get; set; }

    [Column("contact_phone")]
    public string? ContactPhone { get; set; }

    [Column("logo_url")]
    public string? LogoUrl { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("plan_type")]
    public CompanyPlanType PlanType { get; set; } = CompanyPlanType.Basic;

    [Column("max_users")]
    public int MaxUsers { get; set; } = 10;

    [Column("max_whatsapp_numbers")]
    public int MaxWhatsappNumbers { get; set; } = 1;

    [Column("max_ai_agents")]
    public int MaxAiAgents { get; set; } = 1;

    // Navigation properties
    [Column("company_group_id")]
    [Required]
    public Guid CompanyGroupId { get; set; }
    
    [ForeignKey("CompanyGroupId")]
    public virtual CompanyGroup CompanyGroup { get; set; } = null!;

    public virtual ICollection<Department> Departments { get; set; } = new List<Department>();
    public virtual ICollection<WhatsAppInstance> WhatsappInstances { get; set; } = new List<WhatsAppInstance>();
    public virtual ICollection<User> Users { get; set; } = new List<User>();
    public virtual ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();
}